### PR TITLE
 [autogluon][build] AutoGluon 0.4.2 container

### PR DESCRIPTION
**Pull request by gradientsky**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
AutoGluon 0.4.2 containers release

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [x] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/buildspec-0-4.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
+region: &REGION <set-$REGION-in-environment>
+framework: &FRAMEWORK autogluon
+version: &VERSION 0.4.0
```
Comment (saimidu) [inline autogluon/buildspec-0-4.yml:4]: Please make this the 0.4.2 version of the buildspec.yml file. Is there any reason we would want to continue to have AG 0.4.0 images once we start to publish AG 0.4.2?

Comment (gradientsky) [reply autogluon/buildspec-0-4.yml:4]: Good point, no we want `0.4.2` to be the new `0.4`. Will update.

File: autogluon/training/docker/0.4/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -1,179 +1,47 @@
...
-WORKDIR /
+ARG AUTOGLUON_VERSION=0.4.2
+ARG TORCH_VER=1.10.2+cpu
+ARG TORCH_VISION_VER=0.11.3+cpu
 
-RUN pip install --no-cache-dir -U \
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools==59.5.0 wheel \
+ && pip install --no-cache-dir -U \
+    numpy numba \
```
Comment (saimidu) [inline autogluon/training/docker/0.4/py3/Dockerfile.cpu:28]: Does numpy need an upgrade to latest from the version installed on the mxnet DLC?
https://github.com/aws/deep-learning-containers/blob/e24629a4ac2c6c75880b211cc49d822a2c050b5e/mxnet/training/docker/1.9/py3/Dockerfile.cpu#L98

File: autogluon/inference/docker/0.4/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -20,20 +23,27 @@ RUN apt-get update \
 
 ARG TORCH_VER=1.10.2+cpu
 ARG TORCH_VISION_VER=0.11.3+cpu
-ARG AUTOGLUON_VERSION=0.4.0
+ARG AUTOGLUON_VERSION=0.4.2
 
-RUN pip --no-cache-dir install --upgrade --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
- && pip --no-cache-dir install --upgrade wheel setuptools \
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools==59.5.0 wheel \
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply autogluon/inference/docker/0.4/py3/Dockerfile.cpu:29]: We have to pin it because the latest version broke `pip` and was causing installation issues.

File: autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get update \
+ && apt-get -y upgrade \
+ && apt-get autoremove -y \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+ARG AUTOGLUON_VERSION=0.4.2
+
+RUN pip install --no-cache-dir -U --trusted-host pypi.org --trusted-host files.pythonhosted.org pip \
+ && pip install --no-cache-dir -U setuptools==59.5.0 wheel \
```
[the remark this answers is not in the document]
Comment (gradientsky) [reply autogluon/inference/docker/0.4/py3/cu112/Dockerfile.gpu:17]: see above

**Review by saimidu — APPROVED** [review]
Approved. Please revert the toml file changes as soon as the tests succeed, and let's go ahead and merge this PR.

**Comment by gradientsky** [comment]
Tests passed, reverted toml config

**Review by saimidu — APPROVED** [review]
